server/channels/app/bot_test.go

```diff
@@ -112,6 +112,82 @@ func TestCreateBot(t *testing.T) {
 	})
 }
 
+func TestEnsureBot(t *testing.T) {
+	t.Run("ensure bot should pass if already exist bot user", func(t *testing.T) {
+		th := Setup(t).InitBasic()
+		defer th.TearDown()
+		pluginId := "pluginId"
+
+		PluginErr := th.App.SetPluginKey(pluginId, "key", []byte("test"))
+		assert.Nil(t, PluginErr)
+
+		botID1, err1 := th.App.EnsureBot(th.Context, pluginId, &model.Bot{
+			Username:    "username",
+			Description: "a bot",
+			OwnerId:     th.BasicUser.Id,
+		})
+		require.NoError(t, err1)
+
+		getBot, getBotErr := th.App.GetBot(th.Context, botID1, true)
+		require.Nil(t, getBotErr)
+		assert.Equal(t, "username", getBot.Username)
+		assert.Equal(t, "a bot", getBot.Description)
+
+		botID2, err2 := th.App.EnsureBot(th.Context, pluginId, &model.Bot{
+			Username:    "another_username",
+			Description: "another bot",
+			OwnerId:     th.BasicUser.Id,
+		})
+		require.NoError(t, err2)
+		require.True(t, botID1 == botID2)
```

Comment:
`require.Equal` is the idiomatic way to compare things in tests.
```suggestion
		require.Equal(t, botID1, botID2)
```